src/src/Helpers/Networking.cpp

```diff
@@ -1544,7 +1544,31 @@ int http_authenticate(const String& logIdentifier,
     event += '=';
     event += httpCode;
     eventQueue.addMove(std::move(event));
+    #if FEATURE_THINGSPEAK_EVENT
+      // Generate event with the response of a 
+      // "last-value-of-field" tingspeak request (https://de.mathworks.com/help/thingspeak/readlastfieldentry.html)
+      // e.g. (sendToHTTP,api.thingspeak.com,80,/channels/143789/fields/5/last)
+      // where first eventvalue is the channel number, the second the field number
+      // and the third is the value received by the request
+      // Example of received reply: "HTTP : SendToHTTP api.thingspeak.com GETHTTP code: 200 Received reply: 24.2"
+      // Example of the event: "EVENT: ThingspeakReply=1637928,5,24.2"
+      //                                                  ^    ^   ^
+      //                                   channel number ┘    |   └ received value
+      //                                                   field number 
+      // In rules you can grep the reply by "On ThingspeakReply Do ..."
+      if (httpCode == 200 && equals(host, F("api.thingspeak.com")) && uri.endsWith(F("/last"))) {
+        String revent = F("ThingspeakReply");
+        revent += '=';
+        revent += parseString(uri.c_str(), 2, '/');; //get the channel number
```

Comment:
No need to pass in a C-pointer, as `parseString()` can handle the `String` perfectly fine.

And I wonder if building this string would benefit from using `strformat()`, like this:
```
eventQueue.add(strformat(F("ThingspeakReply=%s,%s,%s"),
                         parseStringKeepCase(uri, 2, '/').c_str(),
                         parseStringKeepCase(uri, 4, '/').c_str(),
                         http.getString().substring(0, 21).c_str()));
```
(Using strformat with %s the `.c_str()` _is_ required...)
(And you also doubled a few of these: `;;`)

Reply:
> (And you also doubled a few of these: ;;)

Yes i saw it and immediatly forgot about it when doing other stuff 

> And I wonder if building this string would benefit from using strformat(), like this:

If you say it is better to do it this way then i will change it (this is far beyond my competence... :) )
But why using  `parseStringKeepCase` instead of just `parseString` its only numbers?

Reply:
> If you say it is better

Well, I'd have to have a look at the numbers from this line: `Flash: [=======   ]  74.2% (used 1409525 bytes from 1900544 bytes)`, with and without this modification. Haven't pulled this code local and can't do that quickly, ATM, but you should be able to compare that rather easily.

> its only numbers?

It is in this case, but to make it a little more versatile, and also not perform a `string.toLowerCase()`, as parseString() is actually calling parseStringKeepCase() 😮, it is even a little bit quicker 😉

Reply:
saved 24bytes... thanks a lot!

Reply:
Many smalls will be a big one together 😺

Reply:
so why not doing the same with the http response? saves another 28bytes.

 ```
eventQueue.add(strformat(F("http#%s=%s"),
                        host.c_str(),
                        httpCode));
```

Reply:
That should be `F("http#%s=%d")`, %s = string, %d = decimal (number), %.2f = float (with 2 decimals), etc., regular `C++ sprintf` [formatting](https://www.programiz.com/cpp-programming/library-function/cstdio/sprintf) arguments